geomesa-features/geomesa-feature-common/src/main/scala/org/locationtech/geomesa/features/AbstractSimpleFeature.scala

```diff
@@ -115,6 +110,24 @@ object AbstractSimpleFeature {
     }
     override def setDefaultGeometryProperty(geoAttr: GeometryAttribute): Unit =
       if (geoAttr == null) { setDefaultGeometry(null) } else { setDefaultGeometry(geoAttr.getValue) }
+
+    /**
+     * Mutable feature ID implementation that references back to the parent class. This allows us to
+     * avoid allocating a new object unless `getIdentifier` is called.
```

Comment:
Ok, this is a quick little performance enhancement?  

How did you come to this idea?  I.e., is there a clear need?

Reply:
I started going through to calculate the size of each feature in memory, and there were some local vals that we probably don't need to keep around - i.e. having a FeatureID instance for each feature, keeping the default geometry index in each feature, etc. This way, the FeatureID and  default geometry index will be re-created each time they get called, but generally I would expect that to only happen at most once? So technically it may be slightly slower for certain operations, but it should reduce the memory footprint and avoid some unnecessary calls (e.g. if we never call getFeatureID and just call getID, if we never call getDefaultGeometry).